### ISSUE TITLE
Entry list is refreshed after pressing OK on Edit Entry window.

### ISSUE THREAD

**Issue by leftdisconnected** [kapsiR/HaveIBeenPwnedKeePassPlugin#13]
OS:  Windows 7 64-bit SP2
KeePass v2.49
HaveIBeenPwnedPlugin v0.7.0

** Issue:
Pressing OK on an Edit Entry window to save it causes entry list in the main window to refresh.  Sometimes this occurs instantly, while other times it occurs after 1-5 seconds.  Any applied search filter is lost and the selected entry is often scrolled out of view as KeePass2 switches/refreshes the list to show all entries in the selected entry's Group.

** Notes:
KeePass normally highlights an entry's left-pane Group after selecting a result from a successful search, but it does not refresh the right-pane entry list after editing an entry, even if changes are made; the view remains the same.  The entry list remains as-is, allowing the user to easily perform further actions with that entry.

However, HaveIBeenPwnedPlugin appears to cause the right-pane entry list to be refreshed after a 0-5 second delay and this causes either or both of the following results:

1.  Search result is lost and all entries in the edited entry's Group are sorted with the default (currently set) method.
2.  Selected entry is often scrolled upward out of view _except_ when the entry is near the top or bottom of the default sorted list or all Group entries fit into one view.  This behavior is not exactly predictable, but entries are _generally_ scrolled upward and out of view.

The selected entry remains selected and can be immediately re-opened by using the keyboard Enter key, but the entry is usually out of view.  Since the user typically uses the mouse to press the OK button in the Edit Entry window, they have the mouse in-hand and will need to scroll the entry list to find and re-open the selected entry or to drag-n-drop the username and/or password from that entry.

This behavior has been confirmed in combination with other plugins and when using HaveIBeenPwnedPlugin as the only active plugin.

Thanks for your time regarding this post and for your work in developing and maintaining this useful plugin.  Apologies in advance if this report is inaccurate or falsely attributed.

**Comment by leftdisconnected** [comment]
Update.  I created a clean (portable) KeePass install and added only the HaveIBeenPwnedKeePassPlugin.  The default KeePass settings were manually modified to closely match my primary installation.  A test database was created in my primary KeePass instance and then imported into this Test instance.  Note that my "Primary" instance of KeePass was not running during testing and no global or user KeePass configuration files exist on this system.

The "refresh" did not occur at all during the first run with the new database.  Repeated modifications were made to entries, yet the entry list was never refreshed and search results were always preserved.  I elected to close the database, exit KeePass, and try again.

The issue began occurring after closing and reopening the Test instance of KeePass with the test database and continued to occur after further restarts.  This should confirm that the issue does not require interaction with other plugins or with remnants of previously compiled plugins such as special configuration data.

_However_, though the entry field refresh still occurs in this test, the selected entry is **not** scrolled out of view as it is in my Primary instance of KeePass.  The "Test" behavior is quite tolerable compared with my Primary instance where the entry is reliably scrolled out of view.  I cannot explain why the behavior is different.

As this is a test installation and a test database, I can provide relevant files without disclosing personal information.  As such, I've attached the database and config files to this post.  Simply extract the portable ZIP release of KeePass into this folder to recreate my test environment.  Running keepass.exe should open the included "my_dbs\test.kdbx" database.  Password is "test".  I also have collected a screen capture video of this "benign" behavior with the KeePass_Test instance, available here:

See video description for more info:
https://youtu.be/62jzaJumqm

[KeePass_Test.zip](https://github.com/kapsiR/HaveIBeenPwnedKeePassPlugin/files/7357115/KeePass_Test.zip)

**Comment by leftdisconnected** [comment]
No problem at all and I certainly don't want to waste your time on a minor issue that cannot be readily reproduced.  Spend your time wisely.

It appears to my layman eyes that the issue occurs when/after the plugin checks the HIBP database for the modified entry, but this may be a complex interaction beyond your control.

I have still not managed to reproduce the "scrolled out of view" behavior within the KeePass_Test environment.  I simply observe the loss of the search filter as the selected entry's entire Group is displayed instead.

I did make a recording of my Primary KeePass instance experiencing the original issue, just so that you know what's happening in that environment.  The only difference is that the selected entry is usually moved upward or downward in the view pane, often resulting in that entry being pushed out of the viewable space.  Here's what that looks like:
https://youtu.be/6dSwJGo59lo

If nothing seems obvious, you are welcome to close this minor issue.  The plugin works well enough as-is and I thank you for creating it.

**Comment by kapsiR** [comment]
I can reproduce the issue, where the selection gets lost.  
I have to do some further checks to verify that I can change it [the way I did](https://github.com/kapsiR/HaveIBeenPwnedKeePassPlugin/commit/be6592bf9b6f45ab5a4a98d1da047712be59de59) before releasing a new version.

I uploaded the fixed version here, to help you trying it now.
[v0.7.1-preview.zip](https://github.com/kapsiR/HaveIBeenPwnedKeePassPlugin/files/7360798/v0.7.1-preview.zip)
